pipeline_tools/shared/submission/create_analysis_protocol.py

```diff
@@ -0,0 +1,153 @@
+    }
+
+    def __init__(
+        self,
+        input_uuid,
+        method,
+        pipeline_version,
+        pipeline_type,
+            workspace_version):
+
+        self.input_uuid = input_uuid
+        self.computational_method = method
+        self.protocol_core = {
+            "protocol_id": pipeline_version
+        }
+        self.pipeline_type = pipeline_type
+        self.provenance = {
+            "submission_date": workspace_version,
+            "update_date": workspace_version
+        }
+
+        string_to_hash = json.dumps(self.get_json())
+        entity_id = str(uuid.uuid5(format_map.NAMESPACE, string_to_hash)).lower()
+
+        self.provenance['document_id'] = entity_id
+
+    def __analysis_protocol__(self):
+        return {
+            "computational_method": self.computational_method,
+            "describedBy": self.describedBy,
+            "protocol_core": self.protocol_core,
+            "provenance": self.provenance,
+            "schema_type": self.schema_type,
+            "type": self.type
+        }
+
+    def get_json(self):
+        return self.__analysis_protocol__()
+
+    @property
+    def uuid(self):
+        return self.input_uuid
+
+
+# Entry point for unit tests
+def test_build_analysis_protocol(
+    input_uuid,
+    method,
+    pipeline_version,
+    pipeline_type,
+        workspace_version):
+
+    test_analysis_protocol = AnalysisProtocol(
+        input_uuid,
+        method,
+        pipeline_version,
+        pipeline_type,
+        workspace_version
+    )
+    return test_analysis_protocol.get_json()
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--input_uuid', required=True, help='Input file UUID from the HCA Data Browser')
+    parser.add_argument(
+        '--method',
+        required=True,
+        help='Supposed to be method store url, for now can be url for wdl in skylab, or the name of'
+        ' the analysis workflow.',
+    )
+    parser.add_argument(
+        '--pipeline_version',
+        required=True,
+        help='The version of the pipeline, currently provided by the label of the adapter workflow'
+        ' around the analysis workflow.',
+    )
+    parser.add_argument('--pipeline_type', required=True, help='Type of pipeline(SS2 or Optimus)')
+    parser.add_argument('--workspace_version', required=True, help='Workspace version value i.e. timestamp for workspace')
+
+    args = parser.parse_args()
+
+    analysis_protocol = AnalysisProtocol(
+        args.input_uuid,
+        args.method,
+        args.pipeline_version,
+        args.pipeline_type,
+        args.workspace_version
+    )
+
+    # Get the JSON content to be written
+    analysis_protocol_json = analysis_protocol.get_json()
+
+    # Generate filename based on UUID and version
+    analysis_protocol_filename = f"{analysis_protocol_json['provenance']['document_id']}_{analysis_protocol_json['provenance']['submission_date']}.json"
```

Comment:
turn these two values into @property functions....it basically makes them a getter so you can just do analysis_protocol.document_id